scrape_data/cta_data_downloads.py

```diff
@@ -60,56 +60,166 @@ def save_csv_to_bucket(df: pd.DataFrame, filename: str) -> None:
         .put(Body=csv_buffer.getvalue())
 
 
-def save_sched_daily_summary() -> None:
-    data = sga.GTFSFeed.extract_data(CTA_GTFS)
-    data = sga.format_dates_hours(data)
-    trip_summary = sga.make_trip_summary(data)
-
-    route_daily_summary = (
-        sga.summarize_date_rt(trip_summary)
-    )
-    route_daily_summary['date'] = route_daily_summary['date'].astype(str)
-    route_daily_summary_today = route_daily_summary.loc[route_daily_summary['date'] == today]
-
-    print(f'Saving cta_route_daily_summary_{today}.csv to public bucket')
-    filename = f'schedule_summaries/daily_job/cta_route_daily_summary_{today}.csv'
-    save_csv_to_bucket(
-        route_daily_summary_today,
-        filename=filename
+def save_sched_daily_summary(date_range: List[str] = None) -> None:
+    if date_range is None:
+        date_range = [today]
+        print(f"No date range given. Using {today} only")
+            
+    start_date = pendulum.parse(min(date_range))
+    end_date = pendulum.parse(max(date_range))
+    period = pendulum.period(start_date, end_date)
+    full_date_range = [dt.to_date_string() for dt in period.range('days')]
+    zip_filename_list = [f'cta_schedule_zipfiles_raw/google_transit_{date}.zip'
+                         for date in full_date_range]
+
+    # Check for files in bucket.
+    found_list = keys(
+        csrt.BUCKET_PUBLIC,
+        zip_filename_list
     )
-    print(f'Confirm that {filename} exists in bucket')
-    keys(csrt.BUCKET_PUBLIC, [filename])
+    def confirm_saved_files(file_dict: dict) -> None:
+        for fname in ['csv_filenames', 'zip_filenames']:
+            print('Confirm that ' + ', '.join(file_dict[fname])
+                + ' exist in bucket')  
+            _ = keys(csrt.BUCKET_PUBLIC, file_dict[fname])
+    
+    def extract_date(fname: str) -> str:
+        return fname.split('_')[-1].split('.')[0]
```

Comment:
I trust that this is the right way to parse it, is there a way I can see an example of the type of file that is being parsed though? If it's difficult, don't worry about it, just curious.

Reply:
So let's say you have 

```python
date_range = ['2023-01-01', '2023-05-05']
start_date = pendulum.parse(min(date_range))
end_date = pendulum.parse(max(date_range))
period = pendulum.period(start_date, end_date)
full_date_range = [dt.to_date_string() for dt in period.range('days')]
zip_filename_list = [f'cta_schedule_zipfiles_raw/google_transit_{date}.zip'
                                       for date in full_date_range]
```
An example filename would look like
```python
print(zip_filename_list[0])
cta_schedule_zipfiles_raw/google_transit_2023-01-01.zip
```
Calling `extract_date` gives
```python
print(extract_date(zip_filename_list[0]))
2023-01-01
```
It will split on '_' and take the last entry which is '2023-01-01.zip'. It then splits on '.' and takes the first entry, which is '2023-01-01'.

There's probably a library or regex that would be more robust though.